Add support for --chown flag for ADD/COPY Docker commands

mrfyda opened the pull request:
Closes #1029

lightbend-cla-validator commented:
Hi @mrfyda,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

muuki88 commented:
Thanks a lot for taking the time to work on this. I'll review this when I'm back from vacation  ( 22.10 ) 😍

muuki88 commented on src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala:
```diff
@@ -179,18 +185,34 @@ object DockerPlugin extends AutoPlugin {
-    Cmd("ADD", s"$files /$files")
+
+    dockerVersion match {
+      case Some(DockerVersion(major, minor, _, _)) if major >= 17 && minor >= 9 =>
```
The `if` statement looks buggy to me. 
If the server has version `18.0` it will yield false, which is not want we want, right?
It must be something like `if majro >= (17 && minor >= 9) || major > 17`

At this point I'm fine with putting this into a if statement. If we have more of this special case, we should
extract this in somethin like a `DockerSupport` object:

```scala
object DockerSupport {

   def chownFlag(version: DockerVersion): Boolean = ...

}

```

muuki88 commented:
I updated the branch so all tests should work fine now.

mrfyda commented:
@muuki88 Nice catch. Fixed the condition and extracted it to a separate object.

Also, updated both the spotify docker client implementation and the docs.

ahjohannessen commented:
Thanks @mrfyda 💪🤗

ahjohannessen commented:
@muuki88 Please cut a release :)

muuki88 commented:
Sure sure sure :stuck_out_tongue_winking_eye: 

Merged one last PR today. With sbt cross build and all scripted tests a release always
takes a while, but here it is: `1.3.0`

ahjohannessen commented:
👍

NeQuissimus commented on src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala:
```diff
@@ -80,6 +83,9 @@ object DockerPlugin extends AutoPlugin {
     dockerEntrypoint := Seq("bin/%s" format executableScriptName.value),
     dockerCmd := Seq(),
     dockerExecCommand := Seq("docker"),
+    dockerVersion := Try(Process(dockerExecCommand.value ++ Seq("version --format '{{.Server.Version}}'")).!!)
```
Guys, I don't think this works...
I believe it needs to be `Seq("version", "--format", "'{{.Server.Version}}'")`, otherwise your shell will add another set of ticks around the whole thing and Docker will report that it does not know the command.

NeQuissimus replied:
nvm me, already fixed somewhere else :D

NeQuissimus replied:
#1052 and @mrfyda ftw! :D
